src/nvim/types.h

```diff
@@ -13,4 +13,5 @@ typedef unsigned char char_u;
 // Can hold one decoded UTF-8 character.
 typedef uint32_t u8char_T;
 
+typedef struct expand expand_T;
```

Comment:
The problem seems to be with luajit and autogenerated headers. option.h.generated.h doesn't get type definitions included so luajit doesn't either.
This commit was just an introduction to luajit tests for me, so I made a simple one that wasnt so simple after all.
Anyway should I rework the Pull request more?